### PR TITLE
Remove `Either` documentation page

### LINKED ISSUE #4392: Remove Obsolete `Either` Documentation
The Cats documentation website includes a page on [Either](http://typelevel.org/cats/datatypes/either.html), that IMO is no longer relevant.

Cats no longer supports Scala 2.11, so the unbiased/biased discussion is no longer relevant, and the remaining syntax enrichment for Either is already discussed in the Jumpstart guide. 

The rest is a discussion on typed error handling, but that is not directly Cats related. The specific example also may lead new users down the wrong path, since on the Typelevel stack, I'd expect the application to be implemented on `F[_]` or `IO`, but then we're talking about EitherT, or eschewing typed error handling entirely.

### PR DESCRIPTION
**Pull request by reardonj**
Resolves #4392 

- Remove Either documentation page
- Update FAQ section and links referring to page
- Put `StateT` documentation in the right order

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ a bunch of useful combinators around it. In Scala 2.12.x `Either`
...
+This syntax and the type class instances for `Either` can be imported using `cats.syntax._`, which will also bring in syntactic enrichment and instances for other standard library types, or you can import only the `Either` enrichment with `cats.syntax.either._`.
 
-There are a few minor mismatches between `Xor` and `Either`. For example, in some cases you may need to specify a type parameter for an enrichment method on `Either` (such as `leftMap`) even though it was properly inferred for `Xor`. See the [`Either` section of this guide](datatypes/either.md#either-in-the-small-either-in-the-large) for more information about these issues.
+There are a few minor mismatches between `Xor` and `Either`. For example, in some cases you may need to specify a type parameter for an enrichment method on `Either` (such as `leftMap`) even though it was properly inferred for `Xor`, due `Either` having covariant type parameters.
```
Comment (reardonj) [inline docs/faq.md:42]: I updated this, but is this still an FAQ two major versions later?

File: docs/datatypes/either.md [hunk 2]
```diff
@@ -1,369 +0,0 @@
...
-```scala mdoc:nest
-val either: Either[NumberFormatException, Int] =
-  Either.catchOnly[NumberFormatException]("abc".toInt)
-```
-
-If you want to catch all (non-fatal) throwables, you can use `catchNonFatal`.
-
-```scala mdoc:nest
-val either: Either[Throwable, Int] = Either.catchNonFatal("abc".toInt)
-```
```
Comment (armanbilge) [inline docs/datatypes/either.md:369]: In theory some of this material is still relevant, although we could relocate it to e.g. `ApplicativeError` docs?

Comment (reardonj) [reply docs/datatypes/either.md:369]: Good call. The section actually slotted in nicely.

File: docs/faq.md (hunk 1 above)
Comment (reardonj) [inline docs/faq.md:42]:
```suggestion
There are a few minor mismatches between `Xor` and `Either`. For example, in some cases you may need to specify a type parameter for an enrichment method on `Either` (such as `leftMap`) even though it was properly inferred for `Xor`, due to `Either` having covariant type parameters.
```

**Comment by armanbilge** [comment]
Sorry, one final thought: instead of completely 404ing the page can we put something there to redirect inbound links somewhere more useful?

**Comment by reardonj** [comment]
> Sorry, one final thought: instead of completely 404ing the page can we put something there to redirect inbound links somewhere more useful?

I perused the Laika documentation, but I don't see a way to have a page exist without it showing up in the table of contents. Is there something in the Typelevel site setup to allow such a redirect?

**Comment by armanbilge** [comment]
> but I don't see a way to have a page exist without it showing up in the table of contents.

Jens on Discord:

>  it's `laika.excludeFromNavigation = true` in the config header of md files

Not sure if that's supported in Laika 0.18.x, if not I might need to get the sbt-typelevel upgrade PR in first.

**Review by armanbilge — APPROVED** [review]
Thank you!

**Review by danicheg — APPROVED** [review]
🚀

**Pull request merged**